telegram/_telegramobject.py

```diff
@@ -169,17 +169,19 @@ def _get_attrs(
         """
         data = {}
 
-        if not recursive:
-            try:
-                # __dict__ has attrs from superclasses, so no need to put in the for loop below
-                data.update(self.__dict__)
-            except AttributeError:
-                pass
+        try:
+            # __dict__ has attrs from superclasses, so no need to recompute/duplicate
+            data.update(self.__dict__)
+        except AttributeError:
+            pass
+
+        tuple_data = tuple(data)
         # We want to get all attributes for the class, using self.__slots__ only includes the
         # attributes used by that class itself, and not its superclass(es). Hence, we get its MRO
         # and then get their attributes. The `[:-1]` slice excludes the `object` class
         for cls in self.__class__.__mro__[:-1]:
-            for key in cls.__slots__:  # type: ignore[attr-defined]
+            # add the class's slots with the user defined subclass __dict__ (class has no slots)
+            for key in tuple(cls.__slots__) + tuple_data:  # type: ignore[attr-defined]
```

Comment:
maybe we can use `set` instead? we don't need immutability or ordering, right?

Also `recursive` should be passed to `to_dict` in line 191 below, right? For the tests to cover that, they probably need one more layer of nesting …

Reply:
> we can use set instead? 

I benchmarked this and found tuple to be faster.. (on 3.11)
``` python
>>> timeit.timeit("set(a)|set(b)", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
0.2501877999984572
>>> timeit.timeit("tuple(a)+tuple(b)", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
0.12552469999900495
```

> Also recursive should be passed to to_dict in line 191 

yes, but the default to recurse is `True` so wouldn't really matter in this case

Reply:
> > we can use set instead? 
> 
> I benchmarked this and found tuple to be faster.. on 3.11: 
> ``` python
> >>> timeit.timeit("set(a)|set(b)", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
> 0.2501877999984572
> >>> timeit.timeit("tuple(a)+tuple(b)", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
> 0.12552469999900495
> ```
> 3.10 has a smaller difference - 0.26s for set, 0.18s for tuple

Mh, the more interesting thing IMO would be how fast loops go through tuples and sets. I also just recalled `itertools.chain` - with that we don't even have to built tuples/sets but can just use `chain(__dict__, __slots__)`. Ought to be faster ...

> > Also recursive should be passed to to_dict in line 191 
> 
> yes, but the default to recurse is `True` so wouldn't really matter in this case. So no need for additional tests

A right. Maybe still explicitly pass `recursive=True`? For confused future me? 😄

Reply:
itertools.chain is indeed faster. 
``` python
>>> timeit.timeit("for i in tuple(a)+tuple(b):pass", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
0.198202909000031
>>> timeit.timeit("for i in itertools.chain(a, b):pass", setup="a=(k for k in range(50)); b={k:k for k in range(10)};")
0.18076100700000097
>>> timeit.timeit("for i in set(a)|set(b):pass", setup="a=(k for k in range(50)); b={k:k for k in range(10)}")
0.3299522259999321
```